src/prefect/client/orion.py

```diff
@@ -163,6 +160,39 @@ def __init__(
                 f"Unexpected type {type(api).__name__!r} for argument `api`. Expected 'str' or 'FastAPI'"
             )
 
+        # See https://www.python-httpx.org/advanced/#timeout-configuration
+        httpx_settings.setdefault(
+            "timeout",
+            httpx.Timeout(
+                connect=PREFECT_API_REQUEST_TIMEOUT.value(),
+                read=PREFECT_API_REQUEST_TIMEOUT.value(),
+                write=PREFECT_API_REQUEST_TIMEOUT.value(),
+                pool=PREFECT_API_REQUEST_TIMEOUT.value(),
+            ),
+        )
+        # See https://www.python-httpx.org/advanced/#pool-limit-configuration
+        httpx_settings.setdefault(
+            "limits",
+            httpx.Limits(
+                # We see instability when allowing the client to open many connections at once.
+                # Limiting concurrency results in more stable performance.
+                max_connections=16,
+                max_keepalive_connections=8,
+                # The Prefect Cloud LB will keep connections alive for 30s.
+                # Only allow the client to keep connections alive for 25s.
+                keepalive_expiry=25,
```

Comment:
🥇

Reply:
Should we consider making this configurable or default to a different value if the user is not using Cloud?